src/modals/cpu-pinning/CpuPinningDataProvider.js

```diff
@@ -83,47 +111,93 @@ const getPlacementPolicyHostsIds = (vm) => {
   return hostsIds
 }
 
-const getCpus = (host) => {
-  const topology = host.cpu?.topology
-  if (topology) {
-    return topology.sockets * topology.cores * topology.threads
-  }
-  return 0
-}
-
-const mapVmToModel = (vm) => {
-  return {
-    id: vm.id,
-    name: vm.name,
-    cpuTopology: {
-      sockets: parseInt(getSockets(vm)),
-      cores: parseInt(getCores(vm)),
-      threads: parseInt(getThreads(vm)),
-    },
-    cpuPinningPolicy: vm.cpu_pinning_policy,
-    cpuPinnings: getPinnings(vm),
+const mapVmToPinnedEntity = (vm) => {
+  const pinned = new PinnedEntity(vm.id, vm.name)
+
+  const sockets = +getSockets(vm)
+  const cores = +getCores(vm)
+  const threads = +getThreads(vm)
+  const cpuToPinnedCpuMap = getCpuToPinnedCpuMapForVm(vm)
+
+  if (sockets && cores && threads) {
+    for (let s = 0; s < +sockets; s++) {
+      for (let c = 0; c < +cores; c++) {
+        for (let t = 0; t < +threads; t++) {
+          const cpuId = s * cores * threads + c * threads + t
+          pinned.cpuPinningTopology.add(s, c, cpuId, cpuToPinnedCpuMap.get(cpuId))
+        }
+      }
+    }
   }
+
+  pinned.cpuCount = sockets * cores * threads
```

Comment:
IMHO returning a plain object would be as good - but it's up to you what style you prefer. However in both code styles it makes sense to construct the complete object in one call i.e.
```
return new PinnedEntity({
  vmId, 
  vmName,
  cpuCount,
  ....
});
```

Reply:
OK, I do not like keeping track of x arguments, but if you think it is better...
```
new PinnedEntity(
    vm.id,
    vm.name,
    sockets * cores * threads,
    topology,
    vm.cpu_pinning_policy,
    getCpuPinningString(vm),
  )
```

Reply:
The named-params style addresses that problem - note that `new PinnedEntity({  vmId})` is a short form of `new PinnedEntity({  vmId: vmId})`. It uses object as a map to pass all parameters.